### DISCUSSION TITLE
SQL Error [IX000]: Routine (get_data_type) can not be resolved.

### DISCUSSION THREAD

**Discussion by geertvercamer** [dbeaver/dbeaver#11906] [Q&A]
#### System information: 
- W10 
- since DBeaver version  > 7.2.0

#### Connection specification:
- IBM Informix Dynamic Server Version 10.00.FC5    

#### Describe the problem you're observing:
SQL Error [IX000]: Routine (get_data_type) can not be resolved.
When browsing in the tree to the column level

**Comment by uslss** [comment]
could you check if it is reproducible against 21.0.1?
is there something in logs?

**Comment by geertvercamer** [comment]
yep, still there.

Log:
java.sql.SQLException: Routine (get_data_type) can not be resolved.
at com.informix.util.IfxErrMsg.buildExceptionWithMessage(IfxErrMsg.java:416)
at com.informix.util.IfxErrMsg.buildIsamException(IfxErrMsg.java:401)
at com.informix.jdbc.IfxSqli.addException(IfxSqli.java:3021)
at com.informix.jdbc.IfxSqli.receiveError(IfxSqli.java:3272)
at com.informix.jdbc.IfxSqli.dispatchMsg(IfxSqli.java:2269)
at com.informix.jdbc.IfxSqli.receiveMessage(IfxSqli.java:2194)
at com.informix.jdbc.IfxSqli.executeStatementQuery(IfxSqli.java:1442)
at com.informix.jdbc.IfxResultSet.executeQuery(IfxResultSet.java:183)
at com.informix.jdbc.IfxStatement.executeQueryImpl(IfxStatement.java:902)
at com.informix.jdbc.IfxStatement.executeQuery(IfxStatement.java:184)
at
com.informix.jdbc.IfxDatabaseMetaData.getColumns(IfxDatabaseMetaData.java:3226)
at
org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCDatabaseMetaDataImpl.getColumns(JDBCDatabaseMetaDataImpl.java:1152)
at
org.jkiss.dbeaver.ext.generic.model.meta.GenericMetaModel.prepareTableColumnLoadStatement(GenericMetaModel.java:630)
at
org.jkiss.dbeaver.ext.generic.model.TableCache.prepareChildrenStatement(TableCache.java:82)
at
org.jkiss.dbeaver.ext.generic.model.TableCache.prepareChildrenStatement(TableCache.java:1)
at
org.jkiss.dbeaver.model.impl.jdbc.cache.JDBCStructCache.loadChildren(JDBCStructCache.java:96)
at
org.jkiss.dbeaver.model.impl.jdbc.cache.JDBCStructCache.getChildren(JDBCStructCache.java:221)
at
org.jkiss.dbeaver.ext.generic.model.GenericTableBase.getAttributes(GenericTableBase.java:154)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native
Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown
Source)
at
java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown
Source)
at java.base/java.lang.reflect.Method.invoke(Unknown Source)
at
org.jkiss.dbeaver.model.navigator.DBNDatabaseNode.extractPropertyValue(DBNDatabaseNode.java:879)
at
org.jkiss.dbeaver.model.navigator.DBNDatabaseNode.access$1(DBNDatabaseNode.java:860)
at
org.jkiss.dbeaver.model.navigator.DBNDatabaseNode$PropertyValueReader.run(DBNDatabaseNode.java:922)
at
org.jkiss.dbeaver.model.navigator.DBNDatabaseNode$PropertyValueReader.run(DBNDatabaseNode.java:1)
at
org.jkiss.dbeaver.model.exec.DBExecUtils.tryExecuteRecover(DBExecUtils.java:167)
at
org.jkiss.dbeaver.model.navigator.DBNDatabaseNode.loadTreeItems(DBNDatabaseNode.java:545)
at
org.jkiss.dbeaver.model.navigator.DBNDatabaseNode.loadChildren(DBNDatabaseNode.java:435)
at
org.jkiss.dbeaver.model.navigator.DBNDatabaseNode.getChildren(DBNDatabaseNode.java:209)
at
org.jkiss.dbeaver.model.navigator.DBNDatabaseNode.getChildren(DBNDatabaseNode.java:1)
at
org.jkiss.dbeaver.model.navigator.DBNUtils.getNodeChildrenFiltered(DBNUtils.java:71)
at
org.jkiss.dbeaver.ui.navigator.database.load.TreeLoadService.evaluate(TreeLoadService.java:49)
at
org.jkiss.dbeaver.ui.navigator.database.load.TreeLoadService.evaluate(TreeLoadService.java:1)
at org.jkiss.dbeaver.ui.LoadingJob.run(LoadingJob.java:88)
at org.jkiss.dbeaver.ui.LoadingJob.run(LoadingJob.java:72)
at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:105)
at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)

Met vriendelijke groet,


Geert

**Comment by LonwoLonwo** [comment]
It can be driver issue https://stackoverflow.com/questions/25982045/routine-get-data-type-can-not-be-resolved
What driver version do you use?

**Comment by geertvercamer** [comment]
We are on Informix Dynamic Server 10.00.FC5
Driver is Informix JDBC Driver for Informix Dynamic Server 4.50.JC4W1

When I Install dBeaver 7.2.0 again, I get a request to download driver
files:

   - repo:/drivers/informix/ifxjdb.jar
   - repo:/drivers/informix/ifxlang.jar
   - (repo:/drivers/informix/LICENSE.txt)

This yields version IBM Informix JDBC Driver for IBM Informix Dynamic
Server 3.70.JC2DE

I just managed to reinstall latest dBeaver, remove the required driver from
driver settings and add the above mentioned 2 jars and it works.

People with old database version may have to repeat this action every time
there's an update of dBeaver. Nothing you can solve, I guess?




Met vriendelijke groet,


Geert

**Comment by uslss** [comment]
@geertvercamer drivers could be customized manually accordingly to [guide](https://github.com/dbeaver/dbeaver/wiki/Admin-Manage-Drivers)
